src/main/java/servlets/SearchServlet.java

```diff
@@ -72,14 +80,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       return;
     }
 
-    QueryInformation queryInformation;
-    ImmutableList<Receipt> receipts;
+    ImmutableList<Receipt> receipts = null;
 
-    if (Boolean.parseBoolean(request.getParameter("isNewLoad"))) {
-      receipts = getAllReceipts();
-    } else {
+    if (!checkParameter(request, "isNewSearch") && checkParameter(request, "getNextPage")) {
+      receipts = getNextPageOfReceipts();
+    } else if (!checkParameter(request, "isNewSearch")
+        && checkParameter(request, "getPreviousPage")) {
+      receipts = getPreviousPageOfReceipts();
```

Comment:
Since `getNextPageOfReceipts()` and `getPreviousPageOfReceipts()` have a very similar structure, I think the code would be more clear if they were combined into one function with a boolean parameter that determines whether to set `startCursor()` or `endCursor()`. Also, maybe the `getNextPage` and `getPreviousPage` request parameters could be combined into one. I do see how this might make things more confusing though, so this is more of an option.